TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,599 @@
+            return (360 + AngleUnit.DEGREES.fromUnit(angles.angleUnit, angles.firstAngle));
+        } else {
+            return AngleUnit.DEGREES.fromUnit(angles.angleUnit, angles.firstAngle);
+        }
+    }
+
+    //Experiment with using a field oriented drive instead of a robot oriented drive
+    public void experimentalDrive(double x, double y, double z){
+        /*
+        Guide to motor Powers:
+        Left Front: - (y + x + Z)
+        Right Front: y - x - z
+        Left Rear: y + x - z
+        Right Rear: - (y - x + z)
+         */
+
+        double forward = x; //Invert stick X axis
+        double strafe = y;
+
+        double pi = Math.PI;
+
+        /* Adjust Joystick X/Y inputs by navX MXP yaw angle */
+
+        double gyro_radians = getYaw() * pi/180;
+        double temp = forward * cos(gyro_radians) + strafe * sin(gyro_radians);
+        strafe = -forward * sin(gyro_radians) + strafe * cos(gyro_radians);
+        forward = temp;
+
+        /* At this point, Joystick X/Y (strafe/forward) vectors have been */
+        /* rotated by the gyro angle, and can be sent to drive system */
+
+        this.leftFront.setPower(Range.clip(-(forward+strafe+z),-1,1));
+        this.rightFront.setPower(Range.clip((forward-strafe-z),-1,1));
+        this.leftRear.setPower(Range.clip(-(forward-strafe+z),-1,1));
+        this.rightRear.setPower(Range.clip((forward+strafe-z), -1, 1));
+    }
+
+    public double currentDegrees(double x, double y) {
+        if (((Math.atan2(y, x)) * (180/Math.PI))<0) {
+            return (360 + ((Math.atan2(y, x)) * (180/Math.PI)));
+        } else {
+            return ((Math.atan2(y, x)) * (180/Math.PI));
+        }
+    }
+
+    public double currentZone(double x, double y) {
+        if (((int)(Math.round(currentDegrees(x, y) / 22.5))) <= 15) {
+            return (int)(Math.round(currentDegrees(x, y) / 22.5));
+        } else {
+            return 0;
+        }
+    }
+
+    public void resetEncoders() {
+        leftFront.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        rightFront.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        leftRear.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        rightRear.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        leftFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        leftRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+
+    public boolean straighten(double goal) {
+        if (this.getYaw() > (goal - 1) && this.getYaw() < (goal + 1)) {
+            if (this.getYaw() > (goal - 1)) {
+                drive(Direction.TURNLEFT, 0.2);
+            } else if (this.getYaw() < (goal + 1)) {
+                drive(Direction.TURNRIGHT, 0.2);
+            }
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public void displayEncoders() {
```

Comment:
Nice! I like this. Perhaps we could consider down the road refactoring some of our code to only accepting Telemetry objects in specific diagnostic methods?